### PR TITLE
Fix definition of find methods

### PR DESCRIPTION
**Pull request by tbekolay**
Hi, we've just started using `Motor-Types` and saw a mismatch between these types and the code. These `find` methods immediately return async cursors to be awaited; see https://motor.readthedocs.io/en/stable/api-tornado/gridfs.html#motor.motor_tornado.MotorGridFSBucket.find.

### CODE REVIEW

**Review by L0RD-ZER0 — COMMENTED** [review]
## TLDR; 
This is not a problem with the type definitions. I'll be closing this one off if there are no further issues.

## Proper Answer to this Issue
### Function Implementation:
If you were to take a look at how ``find`` is implemented, you'll find this:

```py
def find(self, *args, **kwargs):
        """..."""  # Removed the docstring because redability
        cursor = self.delegate.find(*unwrap_args_session(args), **unwrap_kwargs_session(kwargs))
        cursor_class = create_class_with_framework(AgnosticCursor, self._framework, self.__module__)

        return cursor_class(cursor, self)
```
Ref: [motor/core.py Line 968 - Line 981](https://github.com/mongodb/motor/blob/master/motor/core.py#L968C5-L981C42)

Here you can see it directly returns the cursor class's instance, no `await`s required to get the cursor.

### But Zero, The docs...
Yeah, I took a look at the docs. Tried the example too. It has an issue, being adeprecation error, they're moving to `async for <cursor>` instead of `fetch_next`, which is a good thing. What I think here is you might be understanding the syntax wrong.

```py
cursor = bucket.find({"filename": "lisa.txt"}, no_cursor_timeout=True)
while (await cursor.fetch_next):
    grid_out = cursor.next_object()
    data = await grid_out.read()
```
Ref: [Motor Docs: Tornado API/MorotGridFSBucket.find(...)](https://motor.readthedocs.io/en/stable/api-tornado/gridfs.html#motor.motor_tornado.MotorGridFSBucket.find)

Here, inside `while` argument i.e. `await cursor.fetch_next`, what's being awaited is not the cursor itself, but rather the result of `fetch_next` getter. Cursor is yet another python object, which doesn't need to be awaited upon.

### What is the use of `async` keyword
In the suggested fix, you have used the `async` keyword before find. It more or less marks that a function returns a coroutine, which is something you do await on. However, in use, find just spits out a cursor, not a coroutine which gives a cursor. Thus, using `async` here would be a wrong type-hint. You can read more at [python docs, async function definition](https://docs.python.org/3/reference/compound_stmts.html#async-def) for further details.

### Possible resolution?
Just use `async for grid_out_object in bucket.find(...)` instead. It's easier, simpler, and less annoying. And it doesn't give you a deprecation warn. It will eliminate the steps of you having to manually fetch next until nothing is left and would directly give you a file-like readable object instead of having to get a `grid_out_object` manually.

### Further issues?
Leave them in the thread. I'll answer them as soon as I can. If there's nothing, I'll be closing the issue.

Have a great day 👍

With Regards, 
@L0RD-ZER0

**Comment by tbekolay** [comment]
Thanks for the detailed explanation @L0RD-ZER0!

> In the suggested fix, you have used the `async` keyword before find. It more or less marks that a function returns a coroutine, which is something you do await on.

I think you might have misread the diff: this PR removes the `async` keyword. The `async def` is what currently resides in `master`.

We ran into this issue because we have a snippet like this:

```
        cursor = db.fs.find()
        return [dict(result) for result in await cursor.to_list()]
```

This runs correctly, but after we installed `Motor-Types` gave us a typing error:

```
$ mypy app
tests/test_app.py:406: error: "Coroutine[Any, Any, AsyncIOMotorGridOutCursor]" has no attribute "to_list"  [attr-defined]
tests/test_app.py:406: note: Maybe you forgot to use "await"?
```

when we added `await` to the `db.fs.find`, the test failed, so it seemed clear that the typing did not match the source. The documentation says that `find` is not an async function, hence why I made this PR to remove the `async` keyword for `find`.

**Comment by L0RD-ZER0** [comment]
Ahhhhh, you're right. My apologies. Give me a moment.

**Review by L0RD-ZER0 — APPROVED** [review]
Yeah, `async` keyword shouldn't be there. My bad. Your help is much appreciated!!

**Comment by L0RD-ZER0** [comment]
Ahhhhh, you're right. My apologies. I indeed misread the diff. Merged!

**Comment by tbekolay** [comment]
Thanks for taking a look so quickly!! And thanks for `Motor-Types`, typing saves us so much dev time and I know maintaining type definitions is heavily on the tedious end of the programming spectrum.

**Comment by L0RD-ZER0** [comment]
I'll update the PyPI with a BugFix version tomorrow morning (roughly 7~ hours from now). Once again, I sincerely apologise for me not handling the issue properly. Thanks again for pointing the error out!!

**Comment by L0RD-ZER0** [comment]
[PyPI listing](https://pypi.org/project/motor-types/) has now been updated. You can update the package using ``pip install -U motor-types`` or the relevant command for your package manager.
